include/openthread/platform/settings.h

```diff
@@ -41,6 +41,10 @@
 extern "C" {
 #endif
 
+#define OT_SETTINGS_KEY_ID_MASK       0x00ff
+
+#define OT_SETTINGS_KEY_VERSION_MASK  0xff00
```

Comment:
I feel that we should update the `otPlatSettings*` APIs to pass version as an argument. That way on read, the upper layer can decide if it wants to keep, migrate or delete the existing persisted setting.